services/diversity/src/services/richness.js

```diff
@@ -12,37 +12,41 @@ module.exports = (RichnessPersistence, restAPI) => {
      *
      * @returns {Object[]} Number of inferred and observed species for the desired group.
      */
-    getNumberSpecies: async (areaType, areaId, group = 'all') => {
-      const data = [
-        {
-          id: 'total',
-          inferred: 40,
-          observed: 4,
-          region_observed: 110,
-          region_inferred: 130,
-        },
-        {
-          id: 'endemic',
-          inferred: 20,
-          observed: 25,
-          region_observed: 120,
-          region_inferred: 100,
-        },
-        {
-          id: 'invasive',
-          inferred: 20,
-          observed: 10,
-          region_observed: 90,
-          region_inferred: 80,
-        },
-        {
-          id: 'threatened',
-          inferred: 15,
-          observed: 20,
-          region_observed: 90,
-          region_inferred: 100,
-        },
-      ];
+    getNumberOfSpecies: async (areaType, areaId, group = 'all') => {
+      const values = await RichnessPersistence.findNumberOfSpecies(areaType, areaId, group);
```

Comment:
The persistence function doesn't receive the group.

But I wonder if it should, should it select only the specified group columns? or should we delete the group parameter completely (from the route) and return always all groups?

@Trjegul84 @cazapatamar what do you think?

Reply:
In the persistence I did not manage to make a filter by group to perform different queries according to "the group" and we decided to have a single query, then that parameter is not used. I also thought that the group parameter was needed in some Frontend interaction, if not then it is not used and it would be better to remove it.

Reply:
The filter would be to select only the involved columns. It would require to construct the query in a more dynamic way, no to do all the grouping in SQL

Reply:
I like better the option we discussed yesterday with @Trjegul84 for persistences. I suggested to create a separate persistence function to each case and join them in the services. It could be a little more laborious but i think is more clear and in the future this persistences could be reused.

The current implementation is highly tied to the data what we need in front. What do you think @erikasv ?

Reply:
Ok, I actually suggested Angela to not implement multiple functions in the persistence :sweat_smile: , but now I see what you mean with the current implementation too tied to the front needs. And I agree, I always prefer something reusable.